hamt/hamt.go

```diff
@@ -380,7 +388,7 @@ func (ds *Shard) rmChild(i int) error {
 	return nil
 }
 
-func (ds *Shard) getValue(ctx context.Context, hv *hashBits, key string, cb func(*shardValue) error) error {
+func (ds *Shard) getValue(ctx context.Context, hv *hashBits, key string, cb func(*Shard) error) error {
```

Comment:
Note to self: this seems like another great example of where we could use the [Walker](https://github.com/ipfs/go-ipld-format/pull/39), generalizing the `Shard` as just another type of node in a DAG, where we could call its `Iterate` method. But, as previously suggested, this may be an unnecessary over-complication. I just don't like seeing the same recursive/switch logic over and over everywhere in the code base.